src/components/Calendar/Calendar.js

```diff
@@ -1,185 +1,142 @@
-							</ul>
-						</div>
-
-						<ul id='weekdays'>
-							<li>Su</li>
-							<li>Mo</li>
-							<li>Tu</li>
-							<li>We</li>
-							<li>Th</li>
-							<li>Fr</li>
-							<li>Sa</li>
-						</ul>
-
-						<ul id='days'></ul>
-					</div>
-				</div>
-			</div>
-		);
-	}
+  constructor(props) {
+    super(props);
+
+
+    const today = new Date();
+    this.state = {
+      dayInt: today.getDate(),
+      month: today.getMonth(),
+      year: today.getFullYear(),
+    };
+  }
+
+  handleClickNext = () => {
+    this.setState(({ month, year }) => {
+      const newYear = year = month === 11 ? year + 1 : year;
+      const newMonth = (month + 1) % 12;
+      
+      return {
+        month: newMonth,
+        year: newYear,
+      };
+    });
+  };
+
+  handleClickPrev = () => {
+    this.setState(({ month, year }) => {
+      const newYear = month === 0 ? year - 1 : year;
+      const newMonth = month === 0 ? 11 : month - 1;
+      
+      return {
+        month: newMonth,
+        year: newYear,
+      };
+    });
+  };
+
+  renderCalendar = () => {
+    const today = new Date();
+    const { events } = this.props;
+    const { dayInt, month, year } = this.state;
+    const firstDay = new Date(year, month).getDay();
+    const totalDays = new Date(year, month + 1, 0).getDate();
+
+    const calendarItems = [];
+
+    // render blank days first 
+    for (let x = 0; x < firstDay; x++) {
+      calendarItems.push(<li key={`blank-${x}`} />);
+    }
+
+    // render rest 
+    for (let day = 1; day <= totalDays; day++) {
+      const isActive = dayInt === day && month === today.getMonth() && year === today.getFullYear();
+
+      // we filter out all the vents on this day.
+      const eventsOnThisDay = events.filter((event) => {
+        const eventDate = event.date.split('-');
+        const eventDay = eventDate[2];
+        const eventMonth = eventDate[1];
+        const eventYear = eventDate[0];
+
+        return (
+          day === eventDay &&
+          (month + 1).toString() === eventMonth &&
+          year === eventYear
+        );
+      });
+
+      calendarItems.push(
+        <li className={`singleDay ${isActive ? 'active' : ''}`} key={`day-${day}`}>
+          {day}
+          {eventsOnThisDay.map((event) => (
+            <Link
+              key={event.eventId}
+              to={`/events/${event.eventId}`}
+            >
+              {event.title}
+            </Link>
+          ))}
+        </li>
+      );
+    }
+
+    return calendarItems;
+  };
+
+  render() {
+    return (
+      <div className='events-calendar-container'>
+        <div className='calendar-content'>
+          <h2>Events Calendar</h2>
+          <div id='calendar'>
+            <div className='month'>
+              <ul>
+                <li id='prev' onClick={this.handleClickPrev}>&#10094;</li>
+                <li id='month'>{months[this.state.month]}</li>
+                <li id='year'>{this.state.year}</li>
```

Comment:
now we can render `year` and `month` directly into the DOM. whenever state changes, React automatically updates these parts of the app.